bc_obps/registration/tests/endpoints/v2/_operations/_operation_id/test_boro_id.py

```diff
@@ -0,0 +1,35 @@
+from registration.models.operation import Operation
+from registration.tests.utils.helpers import CommonTestSetup, TestUtils
+from registration.utils import custom_reverse_lazy
+from model_bakery import baker
+
+
+class TestOperationBoroIdEndpoint(CommonTestSetup):
+    def test_boro_id_endpoint_unauthorized_roles_cannot_patch(self):
+        operation = baker.make_recipe(
+            'utils.operation',
+        )
+        roles = ["cas_pending", "industry_user"]
+        for role in roles:
+            response = TestUtils.mock_patch_with_auth_role(
+                self,
+                role,
+                self.content_type,
+                {},
+                custom_reverse_lazy("operation_boro_id", kwargs={'operation_id': operation.id}),
+            )
+            assert response.status_code == 401
+            assert response.json()['detail'] == "Unauthorized"
+
+    def test_authorized_role_can_patch(self):
+        operation = baker.make_recipe('utils.operation', status=Operation.Statuses.REGISTERED)
+
+        response = TestUtils.mock_patch_with_auth_role(
+            self,
+            "cas_admin",
```

Comment:
any particular reason why "cas_analyst" role wasn't included in this test?

Reply:
In all our permission configs, `cas_analyst` and `cas_admin` are always included together, so there's no way one of them could access an endpoint and another couldn't. (cas-registration/bc_obps/common/permissions.py). But that may not be a great strategy if we do start adding differences :grimacing: I'll add `cas_analyst` to this test